shared/src/business/useCases/practitioners/updatePractitionerUserInteractor.test.js

```diff
@@ -140,4 +170,136 @@ describe('updatePractitionerUserInteractor', () => {
         .calls[0][0].user.email,
     ).toEqual('admissionsclerk@example.com');
   });
+
+  describe('updating email', () => {
+    it('should throw unauthorized error when the logged in user does not have permission to manage emails', async () => {
+      testUser = {
+        role: ROLES.petitioner,
+        userId: 'f7d90c05-f6cd-442c-a168-202db587f16f',
+      };
+
+      await expect(
+        updatePractitionerUserInteractor({
+          applicationContext,
+          user: mockPractitioner,
+        }),
+      ).rejects.toThrow('Unauthorized for updating practitioner user');
+    });
+
+    it('should throw an error when updatedEmail is not available in cognito', async () => {
+      applicationContext
+        .getPersistenceGateway()
+        .isEmailAvailable.mockReturnValue(false);
+
+      await expect(
+        updatePractitionerUserInteractor({
+          applicationContext,
+          user: {
+            ...mockPractitioner,
+            confirmEmail: 'exists@example.com',
+            updatedEmail: 'exists@example.com',
+          },
+        }),
+      ).rejects.toThrow('Email is not available');
+    });
+
+    it('should update the user with the new user.updatedEmail value', async () => {
+      await updatePractitionerUserInteractor({
+        applicationContext,
+        user: {
+          ...mockPractitioner,
+          confirmEmail: 'free-email-to-use@example.com',
+          updatedEmail: 'free-email-to-use@example.com',
+        },
+      });
+
+      expect(
+        applicationContext.getPersistenceGateway().updatePractitionerUser.mock
+          .calls[0][0].user,
+      ).toMatchObject({
+        pendingEmail: 'free-email-to-use@example.com',
+        pendingEmailVerificationToken: expect.anything(),
+      });
+    });
+
+    it("should send the verification email when the user's email is being changed", async () => {
+      await updatePractitionerUserInteractor({
+        applicationContext,
+        user: {
+          ...mockPractitioner,
+          confirmEmail: 'free-email-to-use@example.com',
+          updatedEmail: 'free-email-to-use@example.com',
+        },
+      });
+
+      expect(
+        applicationContext.getUseCaseHelpers().sendEmailVerificationLink.mock
+          .calls[0][0],
+      ).toMatchObject({
+        pendingEmail: 'free-email-to-use@example.com',
+        pendingEmailVerificationToken: expect.anything(),
+      });
+    });
+
+    it("should NOT send the verification email when the user's email is being added for the first time", async () => {
+      mockPractitioner.email = undefined;
+      await updatePractitionerUserInteractor({
+        applicationContext,
+        user: {
+          ...mockPractitioner,
+          confirmEmail: 'free-email-to-use@example.com',
+          updatedEmail: 'free-email-to-use@example.com',
+        },
+      });
+
+      expect(
+        applicationContext.getUseCaseHelpers().sendEmailVerificationLink,
+      ).not.toHaveBeenCalled();
+    });
```

Comment:
We should still be sending out a Cognito "welcome email" though, correct? Could we test for that?

Reply:
You are correct -- but in _that_ case, the system currently relies on cognito configuration alone to send those emails. They are not handled in the JS interactor, and are not something that is testable here.